### PR TITLE
Initial damage logic calculation

### PR DESCRIPTION
**Pull request by g-claudino**
# Description

Creation of initial logic for damage, functions to access pawn information from grid and blueprint logic for input action attack.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: Source/AmebamanBattleNetjob/BattleManager.cpp [hunk 1]
```diff
@@ -62,25 +62,31 @@ void ABattleManager::SpawnEnemies(UWorld *world, const FTransform &transform){
...
-void ABattleManager::PlayerAttackCallback(FIntVector target){
-	AttackOnGrid(EnemyGrid, target);
-}
 
-void ABattleManager::EnemyAttackCallback(FIntVector target){
-	AttackOnGrid(PlayerGrid, target);
+
+void ABattleManager::PlayerAttackCallback(FIntVector target_offset, int damage){
+	FTileData tileData;
+	if (PlayerGrid->GetPawnInfo(Player,tileData)){
```
Comment (lucas1131) [inline Source/AmebamanBattleNetjob/BattleManager.cpp:69]:
```suggestion
	if (PlayerGrid->GetPawnInfo(Player, tileData)){
```

File: Source/AmebamanBattleNetjob/BattleManager.cpp [hunk 2]
```diff
@@ -62,25 +62,31 @@ void ABattleManager::SpawnEnemies(UWorld *world, const FTransform &transform){
...
+	ExecuteAttackOnGrid(PlayerGrid, target_offset, damage);
+}
 
-		// TODO: Create AI controller for this
-		// auto controller = UGameplayStatics::GetSOMEAIController(world, 0);
-		// controller->Possess(enemy);
+void ABattleManager::ExecuteAttackOnGrid(AGrid* grid, FIntVector target, int damage){
+	
+	FTileData gridData;
+	if (grid->GetPawnInfo(target,gridData)){
```
Comment (lucas1131) [inline Source/AmebamanBattleNetjob/BattleManager.cpp:85]:
```suggestion
	if (grid->GetPawnInfo(target, gridData)){
```

File: Source/AmebamanBattleNetjob/Grid.cpp [hunk 3]
```diff
@@ -159,3 +159,25 @@ inline FIntVector AGrid::GridArrayIndexToFIntVector(int idx) {
...
+
+bool AGrid::GetPawnInfo(AGridPawn *pawn, FTileData& result) { 
+	FTileData* data = GridPawnMap[pawn->GetName()];
+	if (data == nullptr){
+		return false;
+	}
+	else{
+		result = *data;
+		return true;
+	}
```
Comment (lucas1131) [inline Source/AmebamanBattleNetjob/Grid.cpp:171]:
```suggestion

	result = *data;
	return true;

```

File: Source/AmebamanBattleNetjob/Grid.cpp [hunk 4]
```diff
@@ -159,3 +159,25 @@ inline FIntVector AGrid::GridArrayIndexToFIntVector(int idx) {
...
+
+bool AGrid::GetPawnInfo(FIntVector gridLocation, FTileData& result) {
+	FTileData* data = GridData[FIntVectorToGridArrayIndex(gridLocation)];
+	if (data == nullptr){
+		return false;
+	}
+	else{
+		result = *data;
+		return true;
+	}
```
Comment (lucas1131) [inline Source/AmebamanBattleNetjob/Grid.cpp:182]:
```suggestion

	result = *data;
	return true;
```

File: Source/AmebamanBattleNetjob/GridPawn.cpp [hunk 5]
```diff
@@ -46,4 +47,32 @@ inline float AGridPawn::Smoothstep(float t){
...
+	else{
+		UE_LOG(
+		LogTemp, 
+		Display, 
+		TEXT("[%s.Damage()] Current HP is %d"), 
+			*this->GetName(), 
+			HP
+			);
+	}
+}
```
Comment (lucas1131) [inline Source/AmebamanBattleNetjob/GridPawn.cpp:76]: remove debug logs

```suggestion
void AGridPawn::Damage(int32 damage){
	HP -= damage;
	if(HP <= 0){
		UE_LOG(LogTemp, Display, TEXT("[%s.Damage()] Player is gone"), *this->GetName());
	}
}
```

File: Source/AmebamanBattleNetjob/GridPawn.h [hunk 6]
```diff
@@ -27,6 +27,9 @@ class AMEBAMANBATTLENETJOB_API AGridPawn : public APawn {
 
 	float GetCollisionHalfHeight();
 
+	UFUNCTION(BlueprintCallable)
+	void Damage(int damage);
```
Comment (lucas1131) [inline Source/AmebamanBattleNetjob/GridPawn.h:31]:
```suggestion
	void Damage(int32 damage);
```

File: Source/AmebamanBattleNetjob/GridPawn.cpp [hunk 7]
```diff
@@ -35,7 +36,7 @@ void AGridPawn::SetupPlayerInputComponent(UInputComponent* PlayerInputComponent)
 }
 
 void AGridPawn::Attack(FIntVector target){
-	BattleManager->PlayerAttackCallback(target);
+	BattleManager->PlayerAttackCallback(target, 1);
```
Comment (lucas1131) [inline Source/AmebamanBattleNetjob/GridPawn.cpp:39]:
```suggestion
	BattleManager->PlayerAttackCallback(target, 1); // TODO refactor this hardcoded so the damage value originates from attack/skill dmg value
```

**Review by lucas1131 — APPROVED** [review]

**Comment by g-claudino** [comment]
Merging commits

**Pull request merged**